torch/distributed/optim/optimizer.py

```diff
@@ -0,0 +1,112 @@
+import torch.distributed.rpc as rpc
+import torch.distributed.autograd as dist_autograd
+
+from collections import defaultdict
+
+class FunctionalOptimizer:
```

Comment:
Is this precise enough as a name? Or could we name it differently, say `LocalOptimizer` or `NonLockingOptimizer`?